sig-and-wg/sig-core/proposals/installer.md

```diff
@@ -0,0 +1,44 @@
+# Kyma Installer - dynamic installation 
+
+Created on 2018-08-23 by Krysian Cieślik (@crabtree) and Jakub Kabza (@jakkab).
+This document is a part of the [Kyma modularization proposal](./modularization.md).
+
+## Status
+
+Proposed on 2018-08-23.
+
+## Motivation
+
+Kyma installation is a complex, time-consuming process, involving multiple components. The procedure is followed every day by a large number of developers who test their features, and thus should easily handle different scenarios without modifications. This is where the dynamic installation comes in handy, unifying, simplifying and automating the process.
+
+## Goal
+
+This proposal aims to enhance developer experience by:
+
+- extending installation customizability 
+- reducing the number of different installation methods
+- avoiding redundant modifications to the Installer component
+
+## Current solution
+
+Kyma installation:
+
+- consists of hard-coded steps
+- uses a variety of installation methods throughout the process, such as custom shell scripts, Helm, and kubectl commands
+- stores overrides in a fixed struct that requires code modifications every time new parameter is introduced
+
+All things considered, the installation process lacks coherence and leaves little room for customization, requiring constant maintenance at the same time.
+
+## Proposal
+
+Suggested modifications:
+
+- Installer no longer supports Bash scripts.
+- In order to increase security, Docker container used to run Installer does not feature Bash at all.
+- Available installation and update methods are limited to Helm and kubectl commands.
+- Installer is provided with a descriptive and comprehensible list of Kyma components, specified in the `kyma-installation` custom resource.
+- The list is declarative. It is for the user to decide which components will be installed.
+- Each component on the list is identified by its name, Namespace and the type of installation method.
+- Installer no longer handles dependencies between components.
```

Comment:
Developer experience will be enhanced even more if Installer will handle dependency management. 
For example EC has a extension concept and automatic resolving dependencies of my component. 
I would like to specify that I want to have RemoteEnvironmentBroker and Installer will deduce that EtcdOperator, ServiceCatalog should be installed before installing REB.

Reply:
I couldn't agree more, IMO Installer should follow the **operator** concept. To be an application with operational knowledge baked in. Will be smarter and more tailored than generic tools. Without dependency management, the installer it's not required at all because installing/updating components in a dummy way can be achieved by using Helm.

Reply:
Dependency management and complex update scenarios will be covered in future. At the moment order and dependency installation is on person preparing for installation.

Reply:
so we are doing a step back? how will the cluster-essential and prerequisite be handled in such scenario?

Reply:
i don't understand what you mean by saying that we are taking step back. cluster-prerequisites was refactored some time ago and rest will be installed the same way as we install them now. atm we have a component list defined in installer custom resource. Only purpose of this list so far is to define wether component will be installed or not, order is hardcoded in installer itself. what we want to achieve is to implement it more dynamically, meaning that user can define new step without changing source code of installer. It also means that order of that list will be important: user defining this list needs to take care that first cluster-essentials will be installed and then core component (depending on cluster-essentials). Also if for example user wants REB then he needs to add core to the list.

Reply:
@mszostok done.